===================================================================
RCS file: /cvs/opie/noncore/settings/packagemanager/oipkg.cpp,v
retrieving revision 1.23
diff -u -u -r1.23 oipkg.cpp
--- packagemanager/oipkg.cpp	5 Jun 2005 22:18:10 -0000	1.23
+++ packagemanager/oipkg.cpp	14 Jan 2006 12:10:24 -0000
@@ -422,6 +422,8 @@
     m_ipkgArgs.force_reinstall = ( m_ipkgExecOptions & FORCE_REINSTALL );
     // TODO m_ipkgArgs.force_remove = ( m_ipkgExecOptions & FORCE_REMOVE );
     m_ipkgArgs.force_overwrite = ( m_ipkgExecOptions & FORCE_OVERWRITE );
+    m_ipkgArgs.force_removal_of_dependent_packages = ( m_ipkgExecOptions & FORCE_RECURSIVE );
+    m_ipkgArgs.verbose_wget = ( m_ipkgExecOptions & FORCE_VERBOSE_WGET );
     m_ipkgArgs.verbosity = m_ipkgExecVerbosity;
     if ( m_ipkgArgs.dest )
         free( m_ipkgArgs.dest );
Index: oipkg.h
===================================================================
RCS file: /cvs/opie/noncore/settings/packagemanager/oipkg.h,v
retrieving revision 1.8
diff -u -u -r1.8 oipkg.h
--- packagemanager/oipkg.h	5 Jun 2005 22:18:10 -0000	1.8
+++ packagemanager/oipkg.h	14 Jan 2006 12:10:24 -0000
@@ -43,6 +43,8 @@
 #define FORCE_REMOVE                            0x0002
 #define FORCE_REINSTALL                         0x0004
 #define FORCE_OVERWRITE                         0x0008
+#define FORCE_RECURSIVE                         0x0010
+#define FORCE_VERBOSE_WGET                      0x0020
 
 class OConfItemList;
 
Index: oipkgconfigdlg.cpp
===================================================================
RCS file: /cvs/opie/noncore/settings/packagemanager/oipkgconfigdlg.cpp,v
retrieving revision 1.20
diff -u -u -r1.20 oipkgconfigdlg.cpp
--- packagemanager/oipkgconfigdlg.cpp	5 Jun 2005 22:41:15 -0000	1.20
+++ packagemanager/oipkgconfigdlg.cpp	14 Jan 2006 12:10:24 -0000
@@ -150,6 +150,10 @@
         options |= FORCE_REMOVE;
     if ( m_optForceOverwrite->isChecked() )
         options |= FORCE_OVERWRITE;
+    if ( m_optForceRecursive->isChecked() )
+        options |= FORCE_RECURSIVE;
+    if ( m_optVerboseWget->isChecked() )
+        options |= FORCE_VERBOSE_WGET;
     m_ipkg->setIpkgExecOptions( options );
     m_ipkg->setIpkgExecVerbosity( m_optVerboseIpkg->currentItem() );
 
@@ -336,9 +340,17 @@
     QWhatsThis::add( m_optForceOverwrite, tr( "Tap here to enable or disable the '-force-overwrite' option for Ipkg." ) );
     layout->addMultiCellWidget( m_optForceOverwrite, 3, 3, 0, 1 );
 
+    m_optForceRecursive = new QCheckBox( tr( "Force Recursive" ), container );
+    QWhatsThis::add( m_optForceRecursive, tr( "Tap here to enable or disable the '-recursive' option for Ipkg." ) );
+    layout->addMultiCellWidget( m_optForceRecursive, 4, 4, 0, 1 );
+
+    m_optVerboseWget = new QCheckBox( tr( "Verbose fetch" ), container );
+    QWhatsThis::add( m_optVerboseWget, tr( "Tap here to enable or disable the '-verbose_wget' option for Ipkg." ) );
+    layout->addMultiCellWidget( m_optVerboseWget, 5, 5, 0, 1 );
+
     QLabel *l = new QLabel( tr( "Information level:" ), container );
     QWhatsThis::add( l, tr( "Select information level for Ipkg." ) );
-    layout->addMultiCellWidget( l, 4, 4, 0, 1 );
+    layout->addMultiCellWidget( l, 6, 6, 0, 1 );
 
     m_optVerboseIpkg = new QComboBox( container );
     QWhatsThis::add( m_optVerboseIpkg, tr( "Select information level for Ipkg." ) );
@@ -346,15 +358,15 @@
     m_optVerboseIpkg->insertItem( tr( "Normal messages" ) );
     m_optVerboseIpkg->insertItem( tr( "Informative messages" ) );
     m_optVerboseIpkg->insertItem( tr( "Troubleshooting output" ) );
-    layout->addMultiCellWidget( m_optVerboseIpkg, 5, 5, 0, 1 );
+    layout->addMultiCellWidget( m_optVerboseIpkg, 7, 7, 0, 1 );
 
     l = new QLabel( tr( "Package source lists directory:" ), container );
     QWhatsThis::add( l, tr( "Enter the directory where package source feed information is stored." ) );
-    layout->addMultiCellWidget( l, 6, 6, 0, 1 );
+    layout->addMultiCellWidget( l, 8, 8, 0, 1 );
 
     m_optSourceLists = new QLineEdit( container );
     QWhatsThis::add( m_optSourceLists, tr( "Enter the directory where package source feed information is stored." ) );
-    layout->addWidget( m_optSourceLists, 7, 0 );
+    layout->addWidget( m_optSourceLists, 9, 0 );
 
     QPushButton *btn = new QPushButton( Opie::Core::OResource::loadPixmap( "folder", Opie::Core::OResource::SmallIcon ),
                                         QString::null, container );
@@ -362,7 +374,7 @@
     btn->setMaximumWidth( btn->height() );
     QWhatsThis::add( btn, tr( "Tap here to select the directory where package source feed information is stored." ) );
     connect( btn, SIGNAL(clicked()), this, SLOT(slotOptSelectSourceListsPath()) );
-    layout->addWidget( btn, 7, 1 );
+    layout->addWidget( btn, 9, 1 );
 
     layout->addItem( new QSpacerItem( 1, 1, QSizePolicy::Minimum, QSizePolicy::Expanding ) );
 }
@@ -433,6 +445,10 @@
         m_optForceRemove->setChecked( true );
     if ( options & FORCE_OVERWRITE )
         m_optForceOverwrite->setChecked( true );
+    if ( options & FORCE_RECURSIVE )
+        m_optForceRecursive->setChecked( true );
+    if ( options & FORCE_VERBOSE_WGET )
+        m_optVerboseWget->setChecked( true );
 
     m_optVerboseIpkg->setCurrentItem( m_ipkg->ipkgExecVerbosity() );
 }
Index: oipkgconfigdlg.h
===================================================================
RCS file: /cvs/opie/noncore/settings/packagemanager/oipkgconfigdlg.h,v
retrieving revision 1.10
diff -u -u -r1.10 oipkgconfigdlg.h
--- packagemanager/oipkgconfigdlg.h	5 Jun 2005 22:18:10 -0000	1.10
+++ packagemanager/oipkgconfigdlg.h	14 Jan 2006 12:10:24 -0000
@@ -96,6 +96,8 @@
     QCheckBox *m_optForceReinstall; // Force reinstall ipkg option checkbox
     QCheckBox *m_optForceRemove;    // Force remove ipkg option checkbox
     QCheckBox *m_optForceOverwrite; // Force overwrite ipkg option checkbox
+    QCheckBox *m_optForceRecursive; // Force recursive ipkg option checkbox
+    QCheckBox *m_optVerboseWget;    // Force verbose_wget ipkg option checkbox
     QComboBox *m_optVerboseIpkg;    // Ipkg verbosity option selection
     QLineEdit *m_optSourceLists;    // Ipkg source lists destination directory
 
